core/web/webauthn_controller.go

```diff
@@ -55,14 +56,14 @@ func (c *WebAuthnController) BeginRegistration(ctx *gin.Context) {
 }
 
 func (c *WebAuthnController) FinishRegistration(ctx *gin.Context) {
-	orm := c.App.SessionORM()
-	user, err := orm.FindUser()
-	if err != nil {
-		c.App.GetLogger().Errorf("error finding user: %s", err)
-		jsonAPIError(ctx, http.StatusInternalServerError, fmt.Errorf("failed to obtain current user record: %+v", err))
+	user, ok := auth.GetAuthenticatedUser(ctx)
+	if !ok {
+		c.App.GetLogger().Errorf("failed to obtain current user from context")
```

Comment:
Is this an invariant violation? if so, consider using AssumptionViolation. If expected, downgrade to warn. See logger level documentation (I would suggest revising all added logs for compliance)

Reply:
It would be an assumption error - there are a couple different points across the handlers that assume that since we are at that point in the call stack then the gin context should have been set - a coding error like that would be warn?

Reply:
Dang, `AssumptionViolation` isn't part of the interface of the App.logger. I think `.Errorf` makes sense here

Reply:
There is a `SugaredLogger` wrapper with `AssumptionViolation*` helper methods: `logger.Sugared(lggr).AssumptionViolationf`

Reply:
Ah nice, updated!